lib/pbench/server/api/resources/datasets_list.py

```diff
@@ -254,19 +463,20 @@ def filter_query(
                 else:
                     try:
                         c = getattr(Dataset, second)
-                        try:
-                            is_str = c.type.python_type is str
-                        except NotImplementedError:
-                            is_str = False
-                        if is_str:
-                            column = c
-                        else:
+                        if vtype == "str" and not isinstance(c.type, String):
                             column = cast(getattr(Dataset, second), String)
+                        else:
+                            if not isinstance(c.type, TYPES[vtype].sqltype):
+                                raise APIAbort(
+                                    HTTPStatus.BAD_REQUEST,
+                                    f"Type {vtype} of value {value} is not compatible with dataset column {c.name}",
+                                )
```

Comment:
Perhaps this is splitting hairs (or just moot), but are the arguments in line 469 in the right order?  (If so, then is the statement in line 472 reporting the relationship correctly?...I feel like like one of them is backwards relative to the other, but I could just be confused.)

Reply:
Column `c` has type `c.type`, isn't an instance of the SQL type in `TYPES[vtype]`, which means that `value` (which has the Python type `vtype` that's equivalent to the SQL `TYPES[vtype].sqltype`) isn't compatible. I thought the sentence was fairly straightforward, but now I get to second-guess it.